racket/collects/syntax/private/modcode-noctc.rkt

```diff
@@ -91,6 +91,15 @@
            (and (file-exists? p) p))
          (apply build-path (reroot-path* base (car roots)) args))]))
 
+(define (get-metadata-path/multiple-sub-paths #:roots roots path0-base sub-paths . args)
+  (define candidates
+    (for/list ([sub-path (in-list (if (list? sub-paths) sub-paths (list sub-paths)))])
+      (apply get-metadata-path #:roots roots path0-base sub-path args)))
+  (or (for/first ([candidate (in-list candidates)]
```

Comment:
I'd fuse the two `for` iterations here into a `for*`, instead of computing candidates that may not be needed. More importantly, to be consistent with the compiled-flie load handler, the nesting should be the other way around: each compiled directory within a root, instead of each root within a compiled directory.

Reply:
`get-metadata-path/multiple-sub-paths` is no longer used and can be removed, right?